pod/annotations.go

```diff
@@ -377,18 +395,20 @@ func parseAnnotations(pod *corev1.Pod, pConf *Config) error {
 		}
 	}
 
-	val, ok := annotations[AnnotationKeyPodSchemaVersion]
-	if ok {
-		parsedVal, pErr := strconv.ParseUint(val, 10, 32)
-		if pErr == nil {
-			parsedUint32 := uint32(parsedVal)
-			pConf.PodSchemaVersion = &parsedUint32
-		} else {
-			err = multierror.Append(err, fmt.Errorf("annotation is not a valid uint32 value: %s", AnnotationKeyPodSchemaVersion))
+	for _, an := range uint32Annotations {
+		val, ok := annotations[an.key]
+		if ok {
+			parsedVal, pErr := strconv.ParseUint(val, 10, 32)
+			if pErr == nil {
+				parsedUint32 := uint32(parsedVal)
+				*an.field = &parsedUint32
+			} else {
+				err = multierror.Append(err, fmt.Errorf("annotation is not a valid uint32 value: %s", an.key))
+			}
```

Comment:
Are you sure this wouldn't be easier to understand as (no pointers, just a normal function):

```
pConf.PodSchemaVersion, err = parseUint32AnnotationAndAppendError(AnnotationKeyPodSchemaVersion, err)
pConf.UserEnvVarsStartIndex, err = parseUint32AnnotationAndAppendError(AnnotationKeyPodTitusUserEnvVarsStartIndex, err)
```

Reply:
Yeah, that's a good idea.  I think I'd rather leave it consistent with the other ways of doing it, and then refactor it in a separate PR.